common/buildcraft/lib/block/VanillaRotationHandlers.java

```diff
@@ -119,47 +172,212 @@ private static EnumActionResult rotateLever(World world, BlockPos pos, IBlockSta
+                    return true;
+                } else if (toTry != EnumFacing.UP && toTry != EnumFacing.DOWN) {
+                    return offsetState.getBlockFaceShape(world, offsetPos, toTry) == BlockFaceShape.SOLID && !BlockBCBase_Neptune.isExceptBlockForAttachWithPiston(offsetState.getBlock());
+                }
+                return false;
+            };
+            return rotateAnyTypeManual(world, pos, state, BlockTorch.FACING, ROTATE_TORCH, tester);
         }
+        return EnumActionResult.PASS;
     }
 
     private static EnumActionResult rotateChest(World world, BlockPos pos, IBlockState state, EnumFacing sideWrenched) {
-        return EnumActionResult.PASS;// TODO (Limited rotation for double chests though)
+        if (state.getBlock() instanceof BlockChest) {
+
+            BlockPos otherPos = null;
+            for (EnumFacing facing : EnumFacing.Plane.HORIZONTAL) {
+                BlockPos candidate = pos.offset(facing);
+                if (world.getBlockState(candidate).getBlock() == state.getBlock()) {
+                    otherPos = candidate;
+                    break;
+                }
+            }
+
+            if (otherPos != null) {
+                IBlockState otherState = world.getBlockState(otherPos);
+                EnumFacing facing = state.getValue(BlockChest.FACING);
+                if (otherState.getValue(BlockChest.FACING) == facing) {
+                    world.setBlockState(pos, state.withProperty(BlockChest.FACING, facing.getOpposite()));
+                    world.setBlockState(otherPos, otherState.withProperty(BlockChest.FACING, facing.getOpposite()));
+                    return EnumActionResult.SUCCESS;
+                }
+            }
+
+            return rotateOnce(world, pos, state, BlockChest.FACING, ROTATE_HORIZONTAL);
+        }
+        return EnumActionResult.PASS;
+    }
+
+    private static EnumActionResult rotateTrapDoor(World world, BlockPos pos, IBlockState state, EnumFacing sideWrenched) {
+        if (state.getBlock() instanceof BlockTrapDoor) {
+
+            if (sideWrenched.getAxis().isHorizontal()) {
+                BlockTrapDoor.DoorHalf half = state.getValue(BlockTrapDoor.HALF);
+                if (half == BlockTrapDoor.DoorHalf.TOP) {
+                    half = BlockTrapDoor.DoorHalf.BOTTOM;
+                } else {
+                    half = BlockTrapDoor.DoorHalf.TOP;
+                }
+                world.setBlockState(pos, state.withProperty(BlockTrapDoor.HALF, half));
+                return EnumActionResult.SUCCESS;
+            }
+
+            return rotateOnce(world, pos, state, BlockTrapDoor.FACING, ROTATE_HORIZONTAL);
+        }
+        return EnumActionResult.PASS;
+    }
+
+    private static EnumActionResult rotateStairs(World world, BlockPos pos, IBlockState state, EnumFacing sideWrenched) {
+        if (state.getBlock() instanceof BlockStairs) {
+
+            if (sideWrenched.getAxis().isHorizontal()) {
```

Comment:
Please don't use this - it's a little confusing to use as getting the correct side every time can be annoying. Perhaps try with this instead? (Same applies to trapdoors, and with those the problem is even worse as all of the sides except the main face are small).
````java
private static EnumActionResult rotateStairs(World world, BlockPos pos, IBlockState state, EnumFacing sideWrenched) {
    if (state.getBlock() instanceof BlockStairs) {

        EnumFacing dir = state.getValue(BlockStairs.FACING);

        if (dir == ROTATE_HORIZONTAL.get(0)) {
            BlockStairs.EnumHalf half = state.getValue(BlockStairs.HALF);
            if (half == BlockStairs.EnumHalf.TOP) {
                half = BlockStairs.EnumHalf.BOTTOM;
            } else {
                half = BlockStairs.EnumHalf.TOP;
            }
            state = state.withProperty(BlockStairs.HALF, half);
        }

        return rotateOnce(world, pos, state, BlockStairs.FACING, ROTATE_HORIZONTAL);
    }
    return EnumActionResult.PASS;
}
````

Reply:
Same goes for the door tbh.

Reply:
But I will look in to it